tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -764,103 +764,28 @@ def objective(trial: Trial) -> float:
         (float("inf"), [sign * 0.0]),  # PRUNED (without intermediate values)
     ]
     assert _tpe.sampler._get_observation_pairs(
-        study, ["x"], False, constraints_enabled=constraints_enabled
+        study, ["x"], constraints_enabled=constraints_enabled
     ) == (
         {"x": [5.0, 5.0, 5.0, 5.0]},
         scores,
         expected_violations,
     )
     assert _tpe.sampler._get_observation_pairs(
-        study, ["y"], False, constraints_enabled=constraints_enabled
+        study, ["y"], constraints_enabled=constraints_enabled
     ) == (
         {"y": [None, None, None, None]},
         scores,
         expected_violations,
     )
     assert _tpe.sampler._get_observation_pairs(
-        study, ["z"], False, constraints_enabled=constraints_enabled
-    ) == (
-        {"z": [0, 0, 0, 0]},  # The internal representation of 'None' for z is 0
-        scores,
-        expected_violations,
-    )
-    assert _tpe.sampler._get_observation_pairs(
-        study, ["x"], True, constraints_enabled=constraints_enabled
-    ) == (
-        {"x": [5.0, 5.0, 5.0, 5.0]},
-        scores,
-        expected_violations,
-    )
-    assert _tpe.sampler._get_observation_pairs(
-        study, ["y"], True, constraints_enabled=constraints_enabled
-    ) == (
-        {"y": []},
-        [],
-        [] if constraints_enabled else expected_violations,
-    )
-    assert _tpe.sampler._get_observation_pairs(
-        study, ["z"], True, constraints_enabled=constraints_enabled
+        study, ["z"], constraints_enabled=constraints_enabled
     ) == (
         {"z": [0, 0, 0, 0]},  # The internal representation of 'None' for z is 0
         scores,
         expected_violations,
     )
 
 
-@pytest.mark.parametrize("direction", ["minimize", "maximize"])
-@pytest.mark.parametrize(
-    "constraints_enabled, constraints_func, expected_violations",
-    [
-        (False, None, None),
-        (True, lambda trial: [(-1, -1), (0, -1), (1, -1), (2, -1)][trial.number], [0, 0, 1, 2]),
-    ],
-)
-def test_get_observation_pairs_multi(
```

Comment:
Can we leave this test case and just remove the `True` flag from the call of `get_observation_pairs`?

Reply:
Good catch! Restored that test: https://github.com/optuna/optuna/pull/4079/commits/b9a15da2b6a4a84fbd09ae6d097cdee0fc4767b7.